staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -185,7 +184,6 @@ func TestWatchErrResultNotBlockAfterCancel(t *testing.T) {
 		w.run(false, true)
 		wg.Done()
 	}()
-	w.errChan <- fmt.Errorf("some error")
```

Comment:
don't we have to replace this with a call to sendError or something? I'm not sure what this test is testing if we just remove this and don't add any other error send

Reply:
sendError synchronously puts thing into resultCh and here nothing reads from it so it would hang forever.

So what this test is now doing is ensuring that after calling cancel() the line below, run() call is effectively finishing and not hanging forever.

But I added sendError() in a goroutine to ensure that sendError doesn't send forever either.